app/src/main/java/com/android/voyageur/model/trip/TripRepositoryFirebase.kt

```diff
@@ -2,66 +2,75 @@ package com.android.voyageur.model.trip
 
 import android.util.Log
 import com.google.firebase.firestore.FirebaseFirestore
+import com.google.firebase.firestore.SetOptions
 
 class TripRepositoryFirebase(private val db: FirebaseFirestore) : TripRepository {
-  private val collectionPath = "trips"
+    private val collectionPath = "trips"
 
-  override fun getNewTripId(): String {
-    return db.collection(collectionPath).document().id
-  }
+    override fun getNewTripId(): String {
+        return db.collection(collectionPath).document().id
+    }
 
-  override fun init(onSuccess: () -> Unit) {
-    db.collection(collectionPath)
-        .get()
-        .addOnSuccessListener { onSuccess() }
-        .addOnFailureListener { exception ->
-          Log.e("TripRepositoryFirebase", "Error initializing repository: ", exception)
-        }
-  }
+    override fun init(onSuccess: () -> Unit) {
+        db.collection(collectionPath)
+            .get()
+            .addOnSuccessListener { onSuccess() }
+            .addOnFailureListener { exception ->
+                Log.e("TripRepositoryFirebase", "Error initializing repository: ", exception)
+            }
+    }
 
-  override fun getTrips(onSuccess: (List<Trip>) -> Unit, onFailure: (Exception) -> Unit) {
-    db.collection(collectionPath)
-        .get()
-        .addOnSuccessListener { result ->
-          val trips = result.map { document -> document.toObject(Trip::class.java) }
-          onSuccess(trips)
-        }
-        .addOnFailureListener { exception ->
-          Log.e("TripRepositoryFirebase", "Error getting trips: ", exception)
-          onFailure(exception)
-        }
-  }
+    override fun getTrips(onSuccess: (List<Trip>) -> Unit, onFailure: (Exception) -> Unit) {
+        db.collection(collectionPath)
+            .get()
+            .addOnSuccessListener { result ->
+                val trips = result.mapNotNull { document ->
+                    try {
```

Comment:
what is the point of adding a try / catch? The documents should be parsed without any errors if they are inserted correctly?

Reply:
You're correct in thinking that ideally, documents should be parsed without errors if they are inserted correctly. However, there are still valid reasons for including a try/catch block when handling Firebase Firestore documents:

**1. Data Integrity Issues**
Even though the documents should be inserted correctly, data integrity can still be compromised in various ways:

_Schema Changes_: Over time, our app's data model (Trip in this case) might evolve, and older documents in the Firestore collection might not match the new schema.
_Partial/Corrupted Data:_ It's possible that some documents are missing fields, or that fields have invalid types. For instance, someone could manually modify the Firestore data and introduce errors.

In such cases, attempting to convert a document to the Trip object could result in an exception, which is why the try/catch block ensures graceful handling of such issues.

**2. External Data Changes**

Firebase Firestore allows users to manually interact with the database via the Firestore console or other Firebase tools. If someone manually modifies or inserts data in Firestore without following the correct schema (either intentionally or by mistake), our code might encounter unexpected formats or missing fields.

**3. Resilience**
By using try/catch, we make our more resilient to unexpected data issues. Instead of the entire process failing due to one malformed or invalid document, the app can handle the error gracefully and continue processing the remaining documents.